scipy/optimize/tests/test_optimize.py

```diff
@@ -42,13 +42,34 @@ def der_logit(x):
 
     r = optimize.check_grad(logit, der_logit, x0)
     assert_almost_equal(r, 0)
+    r = optimize.check_grad(logit, der_logit, x0, 
+                            random_projection=True, seed=1234)
+    assert_almost_equal(r, 0)
 
     r = optimize.check_grad(logit, der_logit, x0, epsilon=1e-6)
     assert_almost_equal(r, 0)
+    r = optimize.check_grad(logit, der_logit, x0, epsilon=1e-6,
+                            random_projection=True, seed=1234)
+    assert_almost_equal(r, 0)
 
     # Check if the epsilon parameter is being considered.
     r = abs(optimize.check_grad(logit, der_logit, x0, epsilon=1e-1) - 0)
     assert_(r > 1e-7)
+    r = abs(optimize.check_grad(logit, der_logit, x0, epsilon=1e-1,
+                                random_projection=True, seed=1234) - 0)
+    assert_(r > 1e-7)
+
+    def neg_reduce(x):
+        return -x.sum()
```

Comment:
For a more rigorous test, you should test the gradient for function that does not treat different directions interchangeably.

Reply:
I will try some other non-linear functions. May be that can help. In the above case gradient is constant so the test is not that effective.

Reply:
I have used `(x*sin(x)).sum()` as a function for testing. Let me know if that is fine. Thanks.

Reply:
Yes, this is much better, thanks!